# 1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

In [1]:
from math import factorial, exp


Из условия видно, что мы работаем с биномиальным распределением. Отсюда:
    Вероятность попадания стрелком p  = 0,8
    Вероятность промаха q = 1 – p = 0,2
    Кол-во экспериментов n = 100 выстрелов
    Интересующий исход k = 85 попаданий

In [2]:
def bernulli(n, k, p):
    q = 1 - p
    return ((factorial(n) / (factorial(k) * factorial(n - k)))*(p**k)*(q**(n-k)))

In [3]:
bernulli(100, 85, 0.8)


0.048061793700746355

Ответ: вероятность того, что стрелок попадет в цель ровно 85 раз составляет 4.8%

# 2.Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

В этой задаче мы имеем дело с распредлением Пуассона, т.к. вероятность крайне мала p = 0.0004, количество испытаний велико n = 5000. Среднее количество наступления события λ = p*n 
Интересуют два исхода: m1 = 0 и m2 = 2

In [4]:
def puasson(n, m, p):
    λ = p*n
    return ((λ**m)/factorial(m))*exp(-λ)

Вероятность, что ни одна лампочка не перегорит в первый день:

In [5]:
puasson(5000, 0, 0.0004)

0.1353352832366127

Вероятность, что перегорят ровно 2 лампочки:

In [6]:
puasson(5000, 2, 0.0004)

0.2706705664732254

Ответ: вероятность, что не перегорит ни одна 13.5%, вероятность, что перегорят 2 лампочки 27%

# 3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

Распределение биноминальное. n = 144, k = 70, p = 0.5

In [7]:
bernulli(144, 70, 0.5)

0.06281178035144776

Ответ: вероятность, что из 144 подбрасываний монеты орел выпадет ровно 70 раз составляет 6.3%

# 4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

а) Найдём вероятность, что все мячи белые

Вероятность того, что оба мяча из первого ящика белые:

In [8]:
p1 = 7/10*6/9
p1

0.46666666666666656

Вероятность того, что оба мяча из второго ящика белые:

In [9]:
p2 = 9/11*8/10
p2

0.6545454545454545

Вероятность, что все 4 мяча белые:

In [10]:
p1*p2

0.30545454545454537

б) Найдём вероятность того, что ровно два мяча белые.
Это возможно когда:
1 - 2 белых из первого ящика и 2 не белых из второго
2 - 2 не белых из первого ящика и 2 белых из второго
3 - 1 белый и 1 не белый из первого ящика и 1 белый и 1 не белый из второго 

Вероятность, что 2 не белых вынуты из второго ящика:

In [11]:
p3 = 2/11*1/10
p3

0.01818181818181818

Вероятность события 1 - 2 белых из первого ящика и 2 не белых из второго:

In [12]:
p1*p3

0.008484848484848482

Вероятность, что 2 не белых вынуты из первого ящика:

In [13]:
p4 = 3/10*2/9
p4

0.06666666666666667

Вероятность события 2 - 2 не белых из первого ящика и 2 белых из второго:

In [14]:
p4*p2

0.04363636363636363

Вероятность, что из 1-го ящика вынут 1 белый и 1 не белый:

In [15]:
p5 = 7/10*3/9
p5

0.23333333333333328

Вероятность, что из 2-го ящика вынут 1 белый и 1 не белый:

In [16]:
p6 = 9/11*2/10
p6

0.16363636363636364

Вероятность события 3 - 1 белый и 1 не белый из первого ящика и 1 белый и 1 не белый из второго 

In [17]:
p5*p6

0.03818181818181817

Вероятность, что 2 мяча из 4-х белые:

In [18]:
p1*p3 + p4*p2 + p5*p6

0.09030303030303029

в) Найдём вероятность того, что хотя бы один мяч белый.
Для этого найдём вероятность когда все мячи не белые:

In [19]:
p3*p4

0.0012121212121212121

Вероятность, что хотя бы 1 мяч белый

In [20]:
1 - p3*p4

0.9987878787878788

Ответ: вероятность того, что все мячи белые 30.5%; вероятность того, что ровно два мяча белые 9%; вероятность того, что хотя бы один мяч белый 99.9%